src/TextInput.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import cx from 'classnames';
+
+/**
+ * @module TextInput
+ */
+class TextInput extends React.Component {
+
+	constructor(props) {
+		super(props);
+		this.state = {
+			value: props.value || '',
+		};
+		this.onChange = this.onChange.bind(this);
+	}
+
+	onChange(e) {
+		this.setState({ value: e.target.value });
+	}
+
+	render() {
+		const {
+			name,
+			value,	// eslint-disable-line no-unused-vars
+			label,
+			labelClassName,
+			className,
+			children,
+			error,
+			required,
+			...other
+		} = this.props;
+
+		const classNames = cx(
+			{ 'field--error': error },
+			className
+		);
+
+		const labelClassNames = cx(
+			{ required },
+			labelClassName
+		);
+
+		return (
+			<div>
+				<label className={labelClassNames} htmlFor={other.id}>
+					{label}
+				</label>
+
+				<input type='text'
+					name={name}
+					value={this.state.value}
+					required={required}
+					className={classNames}
+					onChange={this.onChange}
+					{...other} />
+
+				{ this.props.maxLength && <p className='text--caption align--right'>{this.state.value.length} / {this.props.maxLength}</p> }
+
+				{ error && <p className='text--error'>{error}</p> }
+				{children}
+			</div>
+		);
+	}
+}
+
+TextInput.propTypes = {
+	id: React.PropTypes.string,
+	name: React.PropTypes.string.isRequired,
+	error: React.PropTypes.string,
+	label: React.PropTypes.string,
+	labelClassName: React.PropTypes.string,
+	value: React.PropTypes.string,
```

Comment:
up to you, but I've generally been using the convention that any optional 'html native' attributes passed as props don't need to be specified in `propTypes`, but that is totally a subjective call on my part. My reasoning is that web devs are familiar enough with things like `id`, `name`, and `value` that they should expect them to be strings, and there are technically a lot more 'legal' attribute-props that could be passed in here that we shouldn't have to think about validating... totally up to you, and there's minimal cost to validating the props you specifically handle